src/compiler/checker.ts

```diff
@@ -48423,6 +48423,31 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         let current: Node = node;
         while (current) {
             if (isFunctionLikeOrClassStaticBlockDeclaration(current)) {
+                if (node.label) {
+                    const diagnostic = createDiagnosticForNode(node, Diagnostics.Label_0_used_before_declaration, node.label.escapedText.toString());

```

Comment:
You want to use `unescapeLeadingUnderscores`, not `toString`.

Reply:
You're creating this diagnostic at *every level*.

Reply:
> You're creating this diagnostic at every level - that seems wasteful.

What do you mean by that? I'm creating the diagnostic as soon as a isFunctionLikeOrClassLike.. is hit and there's a label assigned to the current node. I don't quite understand what you mean by "at every level" could you explain?

Reply:
My apologies, I misread the code.

Reply:
The `.toString` is still not right here, FWIW.